### PR TITLE
Refactor docs for Step Zero, a K8 cluster

### PR DESCRIPTION
**Pull request by willingc**
Related to #593.

This refactor adds:

- a Step Zero which is configuring your Kubernetes cluster.
    - add directories in source for cloud vendors with existing docs and placeholder directories for those that we still would like docs
    - add placeholder directories with `.gitkeep` so they can be checked in
    - move single page k8 install docs into step zero and break into vendor specific docs
- Decrease the toctree level displayed in "Creating your JupyterHub" as the subitems were not particularly helpful vs. the space on the page they took
- Add a community resources section
    - put community resources (additional resources) here
    - move gallery of deployments used here

What was not done:
- break up the vendor specific shutdown cluster doc
- fix the annoying caption displaying for toctrees on the index page (that will need some css work). We can't remove the caption as it is used in the sidebar nav.
- new content is not added

[Rendered docs](https://test-zerotojh.readthedocs.io/en/cluster-jupythub-doc/index.html)

### CODE REVIEW

**Comment by minrk** [comment]
I like the dedicated step zero pages a lot! It definitely feels nicer than scrolling past all of the instructions for providers that aren't yours.

I think it's still unclear which additional providers we should cover, but there's no harm in placeholders.

File: doc/source/additional-resources.rst [hunk 1]
```diff
@@ -1,18 +1,19 @@
 .. _additional-resources:
 
-Additional resources
-====================
+Community-authored documentation
```
Comment (choldgraf) [inline doc/source/additional-resources.rst]: this is a great idea

File: doc/source/amazon/step-zero-aws.rst [hunk 2]
```diff
@@ -0,0 +1,172 @@
+.. _amazon-aws:
```
Comment (choldgraf) [inline doc/source/amazon/step-zero-aws.rst]: I'm assuming that you haven't added *new* content to the provider-specific sections, so not giving them as close of a look for wording etc, let me know if that's wrong and you'd like a closer look!

File: doc/source/index.rst [hunk 3]
```diff
@@ -11,27 +11,46 @@ hub remotely.
...
-questions, feel free to reach out at the `issues page`_.
+tools may change quickly.
+
+If you have tips or deployments that you would like to share, see
+:ref:`community-resources`. If you see anything that is incorrect
+or have any questions, feel free to reach out at the `issues page`_.
+
+.. _getting-to-zero:
+
+Getting to Step Zero: your Kubernetes cluster
```
Comment (choldgraf) [inline doc/source/index.rst]: I like this wording :-)

File: doc/source/amazon/step-zero-aws.rst (hunk 2 above)
Comment (willingc) [reply doc/source/amazon/step-zero-aws.rst]: Yep. No big changes to provider content except links to images from moving the file.

File: doc/source/index.rst [hunk 4]
```diff
@@ -78,9 +97,37 @@ cloud-based deployments and tips for maintaining your deployment.
...
+This section gives the community a space to provide information on setting
+up, managing, and maintaining JupyterHub.
+
+.. important::
+   We recognize that Kubernetes has many deployment options. As a project team
+   with limited resources to provide end user support, we rely on community
+   members to share their collective Kubernetes knowledge and JupyterHub
+   experiences.
+
+We hope that you will use this section to share deployments with on a variety
```
Comment (choldgraf) [inline doc/source/index.rst]: with/on - we should choose one!

File: doc/source/index.rst [hunk 5]
```diff
@@ -78,9 +97,37 @@ cloud-based deployments and tips for maintaining your deployment.
...
+of infrastructure and for different use cases.
```
Comment (choldgraf) [inline doc/source/index.rst]: does "variety of infrastructure" parse? maybe it should be "variety of deployments"?

Comment (willingc) [reply doc/source/index.rst]: Yeah. I see a typo too. Please open a PR to reword.

**Comment by choldgraf** [comment]
I had a few small comments that shouldn't be blocking on merge (which is good, since this is now merged haha). Feel free to spot-check the individual comments, or we can just take care of these little things over time. I'm finishing up my last day of euro-traveling and conferencing so will have more time for this starting soon.

**Comment by choldgraf** [comment]
(and thanks very much @willingc , I think this is a much-needed refactor)

**Comment by willingc** [comment]
Thanks @choldgraf. Safe travels.

**Comment by willingc** [comment]
That would be fantastic @jzf2101. Would you mind taking care of that?

**Comment by jzf2101** [comment]
I'll also followup with additional suggestions if I have anything I'd like to add in the review

**Pull request merged**